### DISCUSSION TITLE
Question on binding a SUM in MySQL - giving []uint8 but expected uint64

### DISCUSSION THREAD

**Discussion by mrz1836** [aarondl/sqlboiler#921] [Q&A] [answered]
Here is the query:
```golang
if err = queries.Raw(
		`SELECT count(id) as 'active', IFNULL(sum(balance), 0) as 'balance_value' FROM things WHERE  balance>0`
	).Bind(ctx, databaseRead, stats); err != nil {
		return
	}
```

Here is the struct I am binding to:
```golang
type TotalStatistics struct {
	Active           uint64  `json:"active" boil:"active"`
	Balance          float64 `json:"balance"`
	BalanceValue     uint64  `json:"balance_value" boil:"balance_value"`
	Total            uint64  `json:"total" boil:"total"`
	TotalEarnedValue uint64  `json:"total_earned_value" boil:"total_earned_value"`
	TotalEarned      float64 `json:"total_earned"`
}
```

However here is the error I am getting:
```
failed to bind pointers to obj: sql: Scan error on column index 1, name "balance_value": converting driver.Value type []uint8 ("18446744074481836477") to a uint: value out of range"
```

I would love some suggestions if there is a better solution. I am not sure why its not binding as a `uint64`. This was previously working but after the value got too large, it's now not binding correctly.

**Comment by aarondl** [comment, marked as answer]
This seems like there may be some witchcraft inside the MySQL driver that's doing this.

Because in Go there is no real support for uint64 inside the drivers: https://golang.org/pkg/database/sql/driver/#Value I think that it might be automatically converting the value to a string and when it gets to sqlboiler it sees a string (note it's a slice of []uint8 aka []byte) and then it errors. Your best bet is to use a string to ensure that the full range of values is supported by your driver I think.

**Reply by mrz1836** [reply]
Thanks @aarondl for taking the time to answer that - much appreciated ;-)
